libs/zjolt/libs/JoltC/JoltC.cpp

```diff
@@ -1417,6 +1420,81 @@ JPH_Body_GetInverseCenterOfMassTransform(const JPH_Body *in_body, float out_tran
     const JPH::Mat44 m = body->GetInverseCenterOfMassTransform();
     m.StoreFloat4x4(reinterpret_cast<JPH::Float4 *>(out_transform));
 }
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI JPH_AABox
+JPH_Body_GetWorldSpaceBounds(const JPH_Body *in_body)
+{
+    assert(in_body != nullptr);
+    return static_cast<JPH_AABox>(
+        reinterpret_cast<const JPH::Body *>(in_body)->GetWorldSpaceBounds()
+    );
+}
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI JPH_MotionProperties *
+JPH_Body_GetMotionProperties(JPH_Body *in_body)
+{
+    assert(in_body != nullptr);
+    const auto body = reinterpret_cast<JPH::Body *>(in_body);
+    return reinterpret_cast<JPH_MotionProperties *>(body->GetMotionProperties());
+}
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI JPH_MotionProperties *
+JPH_Body_GetMotionPropertiesUnchecked(JPH_Body *in_body)
+{
+    assert(in_body != nullptr);
+    const auto body = reinterpret_cast<JPH::Body *>(in_body);
+    return reinterpret_cast<JPH_MotionProperties *>(body->GetMotionPropertiesUnchecked());
+}
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI uint64_t
+JPH_Body_GetUserData(const JPH_Body *in_body)
+{
+    assert(in_body != nullptr);
+    return reinterpret_cast<const JPH::Body *>(in_body)->GetUserData();
+}
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI void
+JPH_Body_SetUserData(JPH_Body *in_body, uint64_t in_user_data)
+{
+    assert(in_body != nullptr);
+    reinterpret_cast<JPH::Body *>(in_body)->SetUserData(in_user_data);
+}
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI void
+JPH_Body_GetWorldSpaceSurfaceNormal(
+    const JPH_Body *in_body,
+    const JPH_SubShapeID *in_sub_shape_id,
+    const float in_position[3],
+    float out_normal_vector[3]
+)
+{
+    assert(in_body != nullptr);
+    const JPH::Vec3 v = reinterpret_cast<const JPH::Body *>(in_body)->GetWorldSpaceSurfaceNormal(
+        reinterpret_cast<const JPH::SubShapeID &>(in_sub_shape_id),
+        JPH::Vec3(*reinterpret_cast<const JPH::Float3 *>(in_position))
+    );
+    v.StoreFloat3(reinterpret_cast<JPH::Float3 *>(out_normal_vector));
+}
+//--------------------------------------------------------------------------------------------------
+JPH_CAPI JPH_TransformedShape
+JPH_Body_GetTransformedShape(const JPH_Body *in_body)
+{
+    assert(in_body != nullptr);
+    const auto body = reinterpret_cast<const JPH::Body *>(in_body);
+    return static_cast<JPH_TransformedShape>(
+        reinterpret_cast<const JPH::Body *>(in_body)->GetTransformedShape()
+    );
+}
```

Comment:
This returns a copy so:

```cpp
JPH_CAPI JPH_TransformedShape
JPH_Body_GetTransformedShape(const JPH_Body *in_body)
{
    assert(in_body != nullptr);
    const auto body = reinterpret_cast<const JPH::Body *>(in_body);
    return *reinterpret_cast<const JPH_TransformedShape *>(
        &reinterpret_cast<const JPH::Body *>(in_body)->GetTransformedShape()
    );
}
```

Reply:
This doesn't work because it is taking the address of a temporary var on stack. Rewriting like this makes the compiler happy:
```
JPH_CAPI JPH_TransformedShape
JPH_Body_GetTransformedShape(const JPH_Body *in_body)
{
    assert(in_body != nullptr);
    const auto body = reinterpret_cast<const JPH::Body *>(in_body);
    const JPH::TransformedShape transformed_shape = body->GetTransformedShape();
    return *reinterpret_cast<const JPH_TransformedShape *>(&transformed_shape);
}
```

Reply:
I think both ways should do the same thing. But I guess I don't really understand C/C++ haha!